From d9e21986d47c6a238b11dd83c04eafe507d9f500 Mon Sep 17 00:00:00 2001
From: Giuseppe CONDORELLI <giuseppe.condorelli@st.com>
Date: Tue, 26 May 2009 10:03:10 -0400
Subject: [PATCH] stm_fdma: fix fdma platform declaration for 7105, 7111 and 7141

num_resources for device id 1 was missing and in this way fdma_driver_probe
didn't correctly initialize the second fdma device.

Signed-off-by: Giuseppe Condorelli <giuseppe.condorelli@st.com>
Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7105.c |    1 +
 arch/sh/kernel/cpu/sh4/setup-stx7111.c |    1 +
 arch/sh/kernel/cpu/sh4/setup-stx7141.c |    1 +
 3 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7105.c b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
index ceabe03..866bbf6 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7105.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
@@ -251,6 +251,7 @@ static struct platform_device stx7105_fdma_devices[] = {
 	}, {
 		.name		= "stm-fdma",
 		.id		= 1,
+		.num_resources	= 2,
 		.resource = (struct resource[]) {
 			{
 				.start = 0xfe410000,
diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7111.c b/arch/sh/kernel/cpu/sh4/setup-stx7111.c
index 54f46e2..df5f97e 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7111.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7111.c
@@ -138,6 +138,7 @@ static struct platform_device stx7111_fdma_devices[] = {
 	}, {
 		.name		= "stm-fdma",
 		.id		= 1,
+		.num_resources  = 2,
 		.resource = (struct resource[]) {
 			{
 				.start = 0xfe410000,
diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7141.c b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
index ddd0bfd..cdf7850 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7141.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
@@ -367,6 +367,7 @@ static struct platform_device stx7141_fdma_devices[] = {
 	}, {
 		.name		= "stm-fdma",
 		.id		= 1,
+		.num_resources  = 2,
 		.resource = (struct resource[]) {
 			{
 				.start = 0xfe410000,
-- 
1.6.0.6

